vir/simd_benchmarking.h

```diff
@@ -0,0 +1,98 @@
+/* SPDX-License-Identifier: LGPL-3.0-or-later */
+/* Copyright © 2019-2023 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH
+ *                       Matthias Kretz <m.kretz@gsi.de>
+ */
+
+#ifndef VIR_SIMD_BENCHMARKING_H_
+#define VIR_SIMD_BENCHMARKING_H_
+
+#if __cpp_concepts >= 201907
+#include "simd.h"
+
+#define VIR_HAVE_SIMD_BENCHMARKING 1
+
+namespace vir
+{
+  template <typename T>
+    [[gnu::always_inline]] inline void
+    fake_modify_one(T& x)
+    {
+      if constexpr (std::is_floating_point_v<T>)
+	asm volatile("" : "+x"(x));
+      else if constexpr (stdx::is_simd_v<T> || stdx::is_simd_mask_v<T>)
+	{
+#if defined _GLIBCXX_EXPERIMENTAL_SIMD && __cpp_lib_experimental_parallel_simd >= 201803
+	  auto& d = stdx::__data(x);
+	  if constexpr (sizeof(d) <= 8)
+	    asm volatile("" : "+g"(d));
+	  else if constexpr (requires {{d._M_data};})
+	    asm volatile("" : "+x"(d._M_data));
+	  else if constexpr (requires {{d._S_tuple_size};})
+	    stdx::__for_each(d, [](auto, auto& element) {
+	      fake_modify1(element);
+	    });
+	  else
+	    asm volatile("" : "+x"(d));
+#else
+	  // assuming vir::stdx::simd implementation
+	  if constexpr (sizeof(x) < 16)
+	    asm volatile("" : "+g"(x));
+	  else
+	    asm volatile("" : "+x,g,m"(x));
+#endif
+	}
+      else if constexpr (sizeof(x) >= 16)
+	asm volatile("" : "+x"(x));
+      else
+	asm volatile("" : "+g"(x));
+    }
+
+  template <typename... Ts>
+    [[gnu::always_inline]] inline void
+    fake_modify(Ts&... more)
+    { (fake_modify_one(more), ...); }
+
+  template <typename T>
+    [[gnu::always_inline]] inline void
+    fake_read_one(const T& x)
+    {
+      if constexpr (std::is_floating_point_v<T>)
+	asm volatile("" ::"x"(x));
+      else if constexpr (stdx::is_simd_v<T> || stdx::is_simd_mask_v<T>)
+	{
+#if defined _GLIBCXX_EXPERIMENTAL_SIMD && __cpp_lib_experimental_parallel_simd >= 201803
+	  const auto& d = stdx::__data(x);
+	  if constexpr (sizeof(d) <= 8)
+	    asm volatile("" ::"g"(d));
+	  else if constexpr (requires {{d._M_data};})
+	    asm volatile("" ::"x"(d));
+	  else if constexpr (requires {{d._S_tuple_size};})
+	    stdx::__for_each(d, [](auto, const auto& element) {
+	      fake_read1(element);
+	    });
+	  else
```

Comment:
Although this only one command inside the block, it spans more than one line. I would really prefer `{,}`-braces around this block.

Reply:
I prefer it without the braces. The braces would add two more lines, making the function longer and thus harder to capture in one go.